src/include/mallocMC/reservePoolPolicies/CudaSetLimits_impl.hpp

```diff
@@ -47,6 +52,9 @@ namespace mallocMC
             static void resetMemPool(void * p = nullptr)
             {
                 cudaDeviceSetLimit(cudaLimitMallocHeapSize, 8192U);
+                cudaGetLastError(); // cudaDeviceSetLimit() usually fails if any
+                                    // kernel before used ::malloc(), so let's
+                                    // clear the error state
```

Comment:
@psychocoderHPC if you are fine with this, and a `cudaDeviceReset()` after each unit test solves all issues.

Reply:
Yes CUDA device reset is fine but please only in the unit tests so avoid side effects with the user code.

Reply:
Good. Then we are good already ;)